### DISCUSSION TITLE
dune + webpack integration

### DISCUSSION THREAD

**Discussion by ulrikstrid** [ml-in-barcelona/jsoo-react#147] [Show and tell]
I got dune to run webpack for me with this setup, mainly posting it here if someone else finds it interesting/useful. Not that you will need webpack(-cli) in your path to make this work.

webpack.config.js
```js
const path = require("path");

const isProd = process.env.NODE_ENV === "production";

module.exports = (env) => ({
  entry: path.resolve(env.deps),
  mode: isProd ? "production" : "development",
  resolve: {
    modules: [path.resolve(__dirname, "node_modules"), "node_modules"],
  },
  output: {
    path: path.dirname(path.resolve(env.output)),
    filename: path.basename(env.output),
  },
  node: {
    fs: "empty",
    child_process: "empty",
  },
  module: {
    rules: [
      {
        test: /\.css$/i,
        use: ["style-loader", "css-loader"],
      },
    ],
  },
});
```
dune
```lisp
(rule
 (target Index.js)
 (deps app.bc.js)
 (action
  (progn
   (run webpack --mode=production --env.deps=%{deps} --env.output=%{target}))))
```

**Comment by davesnx** [comment]
Thanks @ulrikstrid. That's sweet. Would you like to add it to the template: https://github.com/ml-in-barcelona/jsoo-react-template?

**Reply by ulrikstrid** [reply]
If you think it makes sense I can do it. But it might be an edge case to do this? I wanted to use it together with dream to serve files from there

**Comment by davesnx** [comment]
Running both processes separately I often get an error from webpack just after saving a ml file. Since dune dumps and generates the bc.js each time. Does your rule fix that issue? (Maybe since it's a deps?)

Just curiosity!

**Reply by ulrikstrid** [reply]
Yeah, dune will call webpack after it has finished building the app.bc.js so there is no issue like that. But it might be too slow of a feedback loop for people used to HMR.
